### DISCUSSION TITLE
Help with installation guide

### DISCUSSION THREAD

**Discussion by karlh1996** [AdriaGual/marvel-snap-bot#74] [Q&A] [answered]
Hi, could anyone help with the installation of this, please? This is not my area, and it would be great if someone could assist in setting up the software. I really appreciate any help you can provide.

**Comment by paperc07** [comment, marked as answer]
What do you need help with I been thinking about writing a guide to help people

**Reply by karlh1996** [reply]
To be honest, I video tutorial of the installation would be amazing or step by step pictures. I know both of these are asking a lot from people though. I think it's great work you guys are doing but for someone who doesn't use these software a simple clone this repository and enter repository doesn't mean a lot to me😅 thanks in advance. Karl

**Reply by paperc07** [reply]
ok let me try and put something together in this thread

**Reply by karlh1996** [reply]
That would be great, I will help with feedback or anything I can add to the guide to make it more dumb friendly. Appreciate the quick response also.

**Reply by paperc07** [reply]
let me know what problems you have, I made a ghetto how to

**Comment by paperc07** [comment]
**I WANT TO MAKE THIS KNOWN I TRY AND HELP OUT AS MUCH AS I CAN BUT IF YOU HAVE A PROBLEM AND IM NOT AROUND I LITERALLY DON'T KNOW SHIT ABOUT CODING IM JUST COMPUTER SAVVY AND CHAT GPT HAS LITERALLY WROTE AND TAUGHT ME MOST ALL OF THIS CODING OTHER THEN OPENING THE FILES IN NOTEPAD++ AND FIGURING IT OUT MYSELF SO GO TO https://chat.openai.com/ MAKE AN ACCOUNT AND START ASKING IT ALL TYPES OF SHIT, BUT FOR THE MOST PART HERE IM GOING TO BE UPDATING AND COMING OUT WITH BETTER GUIDES**
1. First you need python (https://www.python.org/ftp/python/3.11.2/python-3.11.2-amd64.exe)

Before moving onto step 2 cause I am a noob and constantly learning try and run CMD and type all these in one by one
 
pip install opencv-python
pip install opencv-contrib-python
pip install numpy
pip install pyautogui
pip install pytesseract

you can use pip freeze command to see what all packages you have for future reference 

2. Get Open CV here (https://opencv.org/releases/) (skip for now you can always go back)
3. you need GIT (https://git-scm.com/download/win) 
4. Make sure you have blue stacks (https://www.bluestacks.com/bluestacks-5.html)
5. Let's go ahead and get ADB TOOLS ---- EXTRACT IT to your marvel snap bot folder wherever that is on your pc here https://developer.android.com/studio/releases/platform-tools EXTRACT IT to your marvel snap bot folder wherever that is on your pc 
<img width="879" alt="image" src="https://user-images.githubusercontent.com/9095222/218587708-639e62e2-fa39-4d17-ab51-327df531d8b3.png">
6. I opened up the git bash app that came once I installed GIT then paste this in git clone https://github.com/AdriaGual/marvel-snap-bot <img width="398" alt="image" src="https://user-images.githubusercontent.com/9095222/218545357-69ca5216-1776-47ab-82a4-78ee7a94d175.png">
7. Next type in git bash cd marvel-snap-bot 
<img width="379" alt="image" src="https://user-images.githubusercontent.com/9095222/218545704-e2e57dae-f732-47aa-8591-964b9de33214.png">
8. Next is pip install -r requirements.txt 
<img width="466" alt="image" src="https://user-images.githubusercontent.com/9095222/218546064-a30d167d-3956-4724-959c-59f4d41949af.png">
9. <code> # Edit the config.py file

adb_path: Absolute path to where you can execute ADB
project_path: Absolute path to the project
images_folder: Absolute path to the images folder
fields_folder: Absolute path to the fields images folder
mana_folder: Absolute path to the mana images folder
turns_folder: Absolute path to the turns folder
data_folder: Absolute path to the card images folder
tmp_path: Absolute path to a tmp folder


<img width="493" alt="image" src="https://user-images.githubusercontent.com/9095222/218547110-a430fdf4-ced0-4db4-aa29-b67bfa55aa5d.png"> </code>
10. Make sure that Bluestacks has the ADB enabled
Settings -> Advanced -> Android Debug Bridge (ADB)
<img width="670" alt="image" src="https://user-images.githubusercontent.com/9095222/218550528-ac2485af-110b-4774-b83e-e203c4f98930.png"> 
11. Check if you have the correct resolution
The script is made to work on 1600x900 resolution
<img width="687" alt="image" src="https://user-images.githubusercontent.com/9095222/218551796-c3bf4308-4459-4fb6-bb2e-72929f5e888d.png">
12. Now assuming everything is done correctly this is the way I do this last part to launch bot. I SHIFT RIGHT CLICK when im inside the marvel-snap-bot folder and select GIT BASH HERE and it opens up this and I type in python start.py 
<img width="448" alt="image" src="https://user-images.githubusercontent.com/9095222/218590950-5ae58a9b-e43f-4bbb-ac95-5edf805260d4.png">


**VARIANTS ARE NOT DETECETED UNLESS YOU MAKE YOUR OWN SCREENSHOTS I WOULD RECOMMEND JUST USING THE REGULAR CARDS ALWAYS**

IF YOU GET THESE PROBLEMS 

1. ModuleNotFoundError: No module named 'cv2' 

1. make sure your config.py file is saved and you installed open cv
2. Also make sure you have ' and not " (except after tools can have it for the adb path)
3. pip install opencv-contrib-python



2. If you get the pip out of date thing in your git bash or whatever open RUN and CMD and type in python -m pip install --upgrade pip

**Reply by therglerfs** [reply]
How do you add it to bluestacks I have the ADP folder from android studio but i have no idea how to add the bot to bluestacks

**Reply by paperc07** [reply]
ahh ok let me add that part to this guide brb

**Reply by therglerfs** [reply]
youre a legend no lie

**Reply by karlh1996** [reply]
![Config](https://user-images.githubusercontent.com/125160437/218586801-f582ec20-60e0-4637-b5f6-9a219be350f5.png)
is this correct?

**Reply by paperc07** [reply]
if that is were you path is also let me tell you how to do the adb im putting it on guide now

**Reply by karlh1996** [reply]
okay ill wait for the next step and I will see if it is correct. I've completed everything else on the guide up until now. Thanks again, I know this must be frustrating to teach a novice over the internet :'(

**Reply by paperc07** [reply]
I didn't know how to do alot of this when bot first came out but im tech savy and know my way around so sorted it all out anyways its up let me know next problem

**Reply by paperc07** [reply]
also do you have discord if so what's your username?

**Reply by paperc07** [reply]
> How do you add it to bluestacks I have the ADP folder from android studio but i have no idea how to add the bot to bluestacks

you don't add it you just add the platform tool's and it comes with ADB and Bluestacks also comes with ADB built in so it recognizes it

**Reply by syllss** [reply]
how to fix SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape

**Comment by karlh1996** [comment]
![config2](https://user-images.githubusercontent.com/125160437/218588221-c01d2962-4001-403e-98d9-48c2d6dfec6d.jpg
This is why i was confused, my folder is missing some stuff?

**Reply by paperc07** [reply]
no that's how it is at first now drag or extract the platform tools folder from the adb download I linked into that folder

**Reply by karlh1996** [reply]
karlh1996#4546
my username for discord

**Reply by paperc07** [reply]
ok I added you Brace Face Baby#4938 is mine

**Comment by AezzeA** [comment]
Traceback (most recent call last):
  File "C:\Users\towny\marvel-snap-bot\start.py", line 65, in <module>
    play_info = info.get_info(
                ^^^^^^^^^^^^^^
  File "C:\Users\towny\marvel-snap-bot\info.py", line 41, in get_info
    cv2.waitKey(10)
cv2.error: OpenCV(4.6.0) D:\a\opencv-python\opencv-python\opencv\modules\highgui\src\window.cpp:1333: error: (-2:Unspecified error) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Cocoa support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function 'cvWaitKey'

i got this error i already follow all the steps i have no clue what this mean

**Reply by AezzeA** [reply]
NVM solved

**Reply by nixtynine** [reply]
I got this: how do you fix this


sirni@NIX MINGW64 /c/marvel-snap-bot (master)
$ python start.py
Traceback (most recent call last):
  File "C:\marvel-snap-bot\start.py", line 1, in <module>
    from utils import android_connection, global_utils
  File "C:\marvel-snap-bot\utils\android_connection.py", line 4, in <module>
    import config
  File "C:\marvel-snap-bot\config.py", line 1
    adb_path = r"C:\Users\paper\marvel-snap-bot\platform-tools
               ^
SyntaxError: unterminated string literal (detected at line 1)

**Reply by paperc07** [reply]
@nixtynine you need to go get the android platform tools and put it in your marvel_snap_bot folder Step 5

**Reply by nixtynine** [reply]
i did, and the error still shows. i deleted and extracted platform-tools but still get the same issue.

**Reply by nixtynine** [reply]
![pt](https://user-images.githubusercontent.com/125677019/219844260-6624d463-e966-4089-8e75-d16a11b09c54.png)

still getting the same error

**Reply by paperc07** [reply]
Delete all that extra stuff, make it say just platform-tools

Better hurry up and report back I'm in bed lol

**Comment by nixtynine** [comment]
![pt2](https://user-images.githubusercontent.com/125677019/219845129-74032613-74c9-403c-a1cf-c2786d9e4dd2.png)
this happened.

**Reply by paperc07** [reply]
Lmao you can't use my syntax you have to use your own hold on 

In the Config.py it needs to be like this for you c:/marvel-snap-bot-master

**Reply by nixtynine** [reply]
![config](https://user-images.githubusercontent.com/125677019/219845525-1e8b19ad-b66c-4aa2-8efa-8b83402b6d9e.png)

what lines should i change and what should i put.
im terrible at this. thanks in advance btw

**Reply by paperc07** [reply]
1 and 2 

adb_path = r"C: \Users \paper \marvel-snap-bot \platform-tools'
project_path = r 'C: \Users \paper \marvel-snap-bot'
images_folder = project_path +
)limages'
fields_
folder = images_folder +
"fields'
mana_folder = images_folder +
" \mana'
turns folder = images_ folder +
"\turns'
data_folder = project_path +
1\\data
tmp_path = project_path + '\\tmp'


Those are mine for my pc lmao your pc isn't named paper 


Make yours 
adb_path = r"C:\marvel-snap-bot-master\platform-tools"
project_path = r'C:\Users\marvel-snap-bot-master'

**Reply by nixtynine** [reply]
![pt3](https://user-images.githubusercontent.com/125677019/219845933-e2dd69af-760f-4ce3-ace1-e58f07d1b354.png)

**Reply by nixtynine** [reply]
oops. thanks. i think its working

how do i choose the deck it should play?

**Reply by paperc07** [reply]
I am getting around to that I just got into screenshotting and trying to figure all that out but I believe it may be able to pick back and forth possibly, but for now just pick the deck you want it to play with and then launch start.py

**Reply by nixtynine** [reply]
what deck would you recommend when using this, is there a thread for that?

Thanks btw. you da man!

**Reply by paperc07** [reply]
lmao welcome to the big leagues, so you have to take screenshots using print_hand_cards.py and I am literally trying to get help on it here #88 but I know im so close im gonna get it figured out any day now, but not only do you have to get the images, you have to save a few of them and make sure there in your  cplist.py file, then you can start messing around with changing priorities and stuff. But like I said as soon as I figure out these images I am making a universal updated cp_list.py for all of us and new users and I will be adding every single image to the data\images folder I think it is and will get that all taken care of as well as rehauling the whole field_list.py and adding all the images for the data\locations folder .... overwhelmed yet? lmao 

you can also check out these #1 #2  #38

**Reply by nixtynine** [reply]
i kinda get the whole picture. sound interesting and exciting thought. thanks. never though a game would introduce mi to these things.
cheers. will try to help anyway i can.

**Reply by paperc07** [reply]
yeah it's fun. I would try using a basic zoo deck for now there's these threads #21 #60 (#15 is a good one)

**Reply by Dice17** [reply]
![marvel snap bot](https://user-images.githubusercontent.com/126554274/221787905-1d7e2e0c-550c-4432-a5df-ead42241c68b.png)
Sorry for bothering but can i know what is the issue here and how to fix it

**Reply by GustaveLM4** [reply]
> ![marvel snap bot](https://user-images.githubusercontent.com/126554274/221787905-1d7e2e0c-550c-4432-a5df-ead42241c68b.png) Sorry for bothering but can i know what is the issue here and how to fix it

same thin here same error same everthing anyone can help

**Reply by Dice17** [reply]
You need to check your port in bluestacks and change it manually in utils > android connection.py and global utils.py.

![Screenshot (35)](https://user-images.githubusercontent.com/126554274/222868034-2873ddb8-6169-426a-a78f-38acfe4a87e9.png)
In this case my port is 11904 so i need to change every port in android connection.py and global utils.py from 5555 to 11904.

**Comment by jonsnow-jpg** [comment]
Hi I am not tech savvy I followed all steps but now the last step is showing this

$ python start.py
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

I have python installed but its in different folder how to correct this

**Comment by pyzoman1** [comment]
![ffdfd](https://user-images.githubusercontent.com/130190204/234626187-1017e377-3898-45e3-8777-ceab69212d44.JPG)
What am i doing wrong?
Sometimes it would open cards, look at them and then wait, and will not start a new turn, just wait for time to run out... Is it just me??? Or it is how it should be?
it is also trying to play cards when it doesnt have any mana... 
also feels like it doesnt know what powers cards have, it would play angela ast card instead of the first...

**Comment by Ailp117** [comment]
Help?
$ python start.py
Traceback (most recent call last):
  File "C:\Users\Seb\Desktop\snap-bot\start.py", line 1, in <module>
    from utils import android_connection, global_utils
  File "C:\Users\Seb\Desktop\snap-bot\utils\android_connection.py", line 4, in <module>
    import config
  File "C:\Users\Seb\Desktop\snap-bot\config.py", line 1
    adb_path = "C:\Users\Seb\Desktop\snap-bot\platform-tools"
                                                             ^
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape

**Comment by YamiNoKanata** [comment]
After i Follow this guide i type
$ python start.py

and now i got this error:

```py
Traceback (most recent call last):
  File "C:\Users\yamin\marvel-snap-bot\start.py", line 1, in <module>
    from utils import android_connection, global_utils
  File "C:\Users\yamin\marvel-snap-bot\utils\android_connection.py", line 4, in <module>
    import config
  File "C:\Users\yamin\marvel-snap-bot\config.py", line 1
    adb_path = 'C:\Users\yamin\marvel-snap-bot\platform-tools'
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape
```

Please help. =)

**Reply by Dice17** [reply]
> After i Follow this guide i type $ python start.py
> 
> and now i got this error:
> 
> ```python
> Traceback (most recent call last):
>   File "C:\Users\yamin\marvel-snap-bot\start.py", line 1, in <module>
>     from utils import android_connection, global_utils
>   File "C:\Users\yamin\marvel-snap-bot\utils\android_connection.py", line 4, in <module>
>     import config
>   File "C:\Users\yamin\marvel-snap-bot\config.py", line 1
>     adb_path = 'C:\Users\yamin\marvel-snap-bot\platform-tools'
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape
> ```
> 
> Please help. =)

You'll get banned for using this, plus SD already put a cap on season pass XP earned so it's not really worth it
